### PR TITLE
Add check for wxUSE_EXTENDED_RTTI in the header.

### PR DESCRIPTION
**Pull request by catalinr**
The sample requires `wxUSE_EXTENDED_RTTI` to be enabled, and that is correctly signaled in the source files. Just avoid the header flooding the output with errors.

### CODE REVIEW

File: samples/xti/codereadercallback.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/////////////////////////////////////////////////////////////////////////////
+﻿/////////////////////////////////////////////////////////////////////////////
```
Comment (vadz) [inline samples/xti/codereadercallback.h]: It looks like this change adds a BOM to the file, which is unwanted. Could you please get rid of it? TIA!

Comment (catalinr) [inline samples/xti/codereadercallback.h]: Right. I wonder how that happened, and how could it be made more obvious in the diff.
Should be fixed now.

**Comment by vadz** [comment]
FWIW we're probably going to remove (unmaintained and never used, AFAIK) XTI anyhow one of these days, so I wouldn't recommend spending much time on it -- unless you intend to bring it back from the limbo in which it spent the last 10+ years...

**Comment by catalinr** [comment]
Not really.
Actually I built it by mistake and took a few minutes to realize what was wrong with it.

**Pull request merged**